catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -150,9 +157,38 @@ public void init() {
   private CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
       throws UnsupportedQueryException, SourceUnavailableException, FederationException {
     QueryRequest request = cqlRequest.createQueryRequest(catalogFramework.getId(), filterBuilder);
-
     Stopwatch stopwatch = Stopwatch.createStarted();
-    QueryResponse response = catalogFramework.query(request);
+
+    List<QueryResponse> responses = Collections.synchronizedList(new ArrayList<>());
+    List<Result> results =
+        ResultIterable.resultIterable(
+                (queryRequest) -> {
```

Comment:
✏️ Might read a little cleaner if you extract the `QueryFunction` here as a local variable.